File: scripts/sce.exs
#!/usr/local/bin/elixir

defmodule Sce do

  def expr1 do
    "return value of expr1."
  end

  def expr2 do
    IO.puts "Not Short-circuit evaluation."
    "return value of expr2."
  end

end

IO.puts Sce.expr1 || Sce.expr2

File: scripts/sce.scala
object Sce {
  def main(args: Array[String]): Unit = {
    println(expr1() || expr2())
  }
  def expr1(): Boolean = {
    return true
  }
  def expr2(): Boolean = {
    println("Not Short-circuit evaluation.")
    return true
  }
}
